### ISSUE TITLE
`nan` matrix for all parameters after training for thousands of steps

### ISSUE THREAD

**Issue by crazylyf** [thtrieu/darkflow#381]
When training on my own dataset, the parameters and loss may become `nan` suddenly. Does anyone encounter similar problem? What's the exact reason?

**Comment by tankienleong** [comment]
May I know which optimizer you are using? I encountered this problem when I used the RMSPROP optimizer but no problem with ADAM optimizer.

**Comment by crazylyf** [comment]
I use the default RMSPROP. I think the problem is related to huge gradients, and gradient clipping  may be a safe way to solve it.

**Comment by andresmafladelgado** [comment]
@crazylyf Hey I have been training with a modified version of this code, however after a lot of steps the nan loss value appeared. Did you find a way to solve this issue?

**Comment by llping** [comment]
@crazylyf @DreadPiratePsyopus  i encountered this problem too, i use RMSPROP, after 60000 steps the gradients became NaN.
i used a very small learning rate 1e-6, but it still no help.  did you solve this problem?
many thanks if you can help.

**Comment by alvinxiii** [comment]
@crazylyf @llping do you have the solution for this issue? I followed suggestions from #482, the problem still persists.
